### PR TITLE
Bytecode gen test gc finalization un flake

### PR DESCRIPTION
**Pull request by jccarrillo**
Cherry-picked from https://github.com/google/guice/pull/981
The https://github.com/google/guice/pull/981 build was failing.

### CODE REVIEW

**Comment by sameb** [comment]
Thanks!  Can you inline the getWeakReference method, similar to how it was before?  Otherwise it's a little hard to follow the test.

**Comment by jccarrillo** [comment]
Will do.

**Comment by jccarrillo** [comment]
@sameb, I made it more inlined with the previous lines. give me your comments. Thanks.
